payjoin/src/core/receive/mod.rs

```
@@ -475,7 +477,34 @@ mod tests {
                 .err()
                 .unwrap(),
             PsbtInputError::from(InvalidScriptPubKey(AddressType::P2wsh))
-        )
+        );
+
+        // p2wsh input with witness data should be supported and should **not** use the provided expected weight
+        let mut dummy_witness = witness::Witness::new();
+        dummy_witness.push(DUMMY32);
+        let weight_from_witness = Weight::from_witness_data_size(dummy_witness.size() as u64);
+        assert_ne!(weight_from_witness, expected_weight);
```

Hmm, overriding an explicitly specified weight seems like unexpected behavior from the perspective of the caller. Maybe it should raise an error instead, if they provide an expected weight when it can be calculated?

---

Good call. Error makes sense to me here